serve_event.go

```diff
@@ -0,0 +1,117 @@
+package wiremock
+
+type ServeEvents struct {
+	ServeEvents []ServeEvent `json:"serveEvents,omitempty"`
+}
+
+type ServeEvent struct {
+	RequestJournalDisabled bool             `json:"requestJournalDisabled,omitempty"`
+	Meta                   Meta             `json:"meta,omitempty"`
+	Requests               []RequestElement `json:"requests,omitempty"`
+
+	Request            ServeEventRequest       `json:"request,omitempty"`
+	StubMapping        StubMapping             `json:"stubMapping,omitempty"`
+	WasMatched         bool                    `json:"wasMatched,omitempty"`
+	Response           PurpleResponse          `json:"response,omitempty"`
+	Timing             Timing                  `json:"timing,omitempty"`
+	ID                 string                  `json:"id,omitempty"`
+	ResponseDefinition ResponseDefinitionClass `json:"responseDefinition,omitempty"`
+}
+
+type Meta struct {
+	Total int64 `json:"total,omitempty"`
+}
+
+type RequestElement struct {
+	Request            ServeEventRequest       `json:"request,omitempty"`
+	StubMapping        StubMapping             `json:"stubMapping,omitempty"`
+	WasMatched         bool                    `json:"wasMatched,omitempty"`
+	Response           PurpleResponse          `json:"response,omitempty"`
+	Timing             Timing                  `json:"timing,omitempty"`
+	ID                 string                  `json:"id,omitempty"`
+	ResponseDefinition ResponseDefinitionClass `json:"responseDefinition,omitempty"`
+}
+
+type ServeEventRequest struct {
+	Headers             RequestHeaders `json:"headers,omitempty"`
+	Method              string         `json:"method,omitempty"`
+	Scheme              string         `json:"scheme,omitempty"`
+	BrowserProxyRequest bool           `json:"browserProxyRequest,omitempty"`
+	QueryParams         Cookies        `json:"queryParams,omitempty"`
```

Comment:
This should be its own type, which is something like this
```go
type Params map[string]Param

type Param struct {
	Key    string `json:"key"`
	Values []string `json:"values"`
}
```
`FormParams` is also missing